MatrixKit/Models/Account/MXKAccountManager.m

```diff
@@ -445,17 +446,24 @@ - (void)setPushDeviceToken:(NSData *)pushDeviceToken withPushOptions:(NSDictiona
             // turn on the Push flag for all accounts
             for (MXKAccount *account in activeAccounts)
             {
-                account.enablePushKitNotifications = YES;
+                [account enablePushKitNotifications:YES success:nil failure:nil];
             }
         }
         else if (![oldToken isEqualToData:pushDeviceToken])
         {
             NSLog(@"[MXKAccountManager][Push] setPushDeviceToken: Update Push device token");
-            
+
+            NSMutableArray<MXKAccount*> *accountsWithPushKitPusher = [NSMutableArray new];
+
             // Delete the pushers related to the old token
             for (MXKAccount *account in activeAccounts)
             {
-                [account deletePushKitPusher];
+                if (account.hasPusherForPushKitNotifications)
```

Comment:
Don't you think you should check `isPushKitNotificationActive` instead of `hasPusherForPushKitNotifications`?

Reply:
`hasPusherForPushKitNotifications` is less restrictive. I think `hasPusherForPushKitNotifications` better manages use cases when user disables notifications permissions.